services/pipenv-helpers.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { InvalidParameter } = require('.')
+
+const isDependency = Joi.object({
+  version: Joi.string().required(),
+}).required()
+
+const isLockfile = Joi.object({
+  _meta: Joi.object({
+    requires: Joi.object({
+      python_version: Joi.string().required(),
```

Comment:
Can we only require this key for the python version badge? Although its in by default, `python_version` is not required. If my `Pipfile` is 

```toml
[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[dev-packages]

[packages]

```

`pipenv lock` will happily produce

```json
{
    "_meta": {
        "hash": {
            "sha256": "ebffa69a1fa192d1cef7cb42ad79231ca976565c5ce371a70160b3048d3cbc06"
        },
        "pipfile-spec": 6,
        "requires": {},
        "sources": [
            {
                "name": "pypi",
                "url": "https://pypi.org/simple",
                "verify_ssl": true
            }
        ]
    },
    "default": {},
    "develop": {}
}
```

which would also prevent the `dependency-version` badge from working.

Unfortunately removing `python_version` is sometimes needed for projects which need to work on more than one python version because `python_version` can't be multiple versions or a range.

Reply:
Sounds good. I'd also rather present a better error for the python version on a project where it's missing. Any idea of a project that looks like that?

If not I could probably write it in a unit test instead.

Reply:
Sure. Here's one: https://github.com/fikovnik/ShiftIt/blob/master/Pipfile.lock